client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,74 @@
+import { Controller } from '@hotwired/stimulus';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath: any;
+  }
+}
+
+/**
+ * Adds the ability to unpack a Telepath object and render it on the controlled element.
+ *
+ * @example
+ * <div
+ *  id="some-id"
+ *  data-controller="w-block"
+ *  data-w-block-data-value='{"some":"json"}'
+ * >
+ * </div>
+ */
+export class BlockController extends Controller<HTMLElement> {
+  static values = {
+    initial: { type: Array, default: [] },
+    error: { type: Object, default: {} },
+    data: { type: Object, default: {} },
+  };
+
+  /** Initial value. */
+  declare initialValue: Array<string>;
+  /** Optional error value. */
+  declare errorValue: object;
+  /** Block data value to unpack with Telepath. */
+  declare dataValue: object;
+
+  connect() {
+    const element = this.element;
+    const id = element.id;
+
+    if (!id) throw new Error('Block element needs an id');
+
+    const telepath = window.telepath;
+
+    if (!telepath) throw new Error("Telepath doesn't exit");
+
+    const output = telepath.unpack(this.dataValue);
+    output.render(element, id, this.initialValue, this.errorValue);
```

Comment:
We would then replace the call here with a JS Spread of the `argumentsValue`.

```suggestion
    output.render(element, id, ...this.argumentsValue);
```